plugins/MauticFocusBundle/Entity/StatRepository.php

```diff
@@ -124,18 +124,22 @@ public function getStatsClickByLead(int $leadId, array $options = []): array
      *
      * @return array<string, mixed>
      */
-    private function getStatsByLeadAndType(int $leadId, string $type, array $options = []): array
+    private function getStatsByLeadAndType(?int $leadId, string $type, array $options = []): array
     {
         $q = $this->getEntityManager()->getConnection()->createQueryBuilder();
 
         $q->from(MAUTIC_TABLE_PREFIX.'focus_stats', 's')
             ->select('s.id, s.lead_id, s.type, s.date_added, f.id as focus_id, f.name as focus_name')
             ->leftJoin('s', MAUTIC_TABLE_PREFIX.'focus', 'f', 's.focus_id=f.id');
 
-        $q->where($q->expr()->and(
-            $q->expr()->eq('s.lead_id', (int) $leadId),
-            $q->expr()->eq('s.type', ':type')
-        ));
+        if ($leadId) {
+            $q->where($q->expr()->and(
+                $q->expr()->eq('s.lead_id', (int) $leadId),
+                $q->expr()->eq('s.type', ':type')
+            ));
+        } else {
+            $q->where($q->expr()->eq('s.type', ':type'));
+        }
```

Comment:
This could be simplified without repeating the type condition.
```suggestion
        $q->where($q->expr()->eq('s.type', ':type'));

        if ($leadId) {
            $q->andWhere($q->expr()->eq('s.lead_id', (int) $leadId));
        } 
```

Reply:
Looks better. Thx!